SLAM/iterative_closest_point/iterative_closest_point.py

```diff
@@ -48,6 +48,10 @@ def icp_matching(previous_points, current_points):
         indexes, error = nearest_neighbor_association(previous_points, current_points)
         Rt, Tt = svd_motion_estimation(previous_points[:, indexes], current_points)
 
+        if dError < 0:  # prevent matrix H changing, exit loop
```

Comment:
dError is only updated in line 60. If we check the value in line 64 with "dError <= EPS", we can check the value is negative or not. So, I think this check is not needed here. What do you think?

Reply:
I put ```if dError < 0``` before the line 58 that changes the state of matrix H. If the error increases, we want to use matrix H from the previous iteration when error was lower. So breaking before line 58 keeps previous iteration transformation.

Reply:
> we want to use matrix H from the previous iteration
Yes. I know what you want. But, I am not sure your change achieve it.
dError is only updated in line 60, so if dError changed to negative, it always break at line 64 without you change (if dError <= EPS.). 
I think this means previous code works  based on what you want.

Reply:
Yes, you are right. Thank you for watching. My bad, I didn't transfered the code correctly, as I did some more modifications for my project.
1. dError should not be absolute, otherwise we cannot catch negative dError.
2. dError assignment goes before ```if dError < 0``` check.
3. Also changed from Converge to Not Converge as when error is increasing it's the second case.